npm/pkg/dataplane/types.go

```diff
@@ -37,19 +37,27 @@ type updateNPMPod struct {
 // todo definitely requires further optimization between the intersection
 // of types, PodMetadata, NpmPod and corev1.pod
 type PodMetadata struct {
-	PodKey   string
-	PodIP    string
-	NodeName string
+	PodKey          string
+	PodIP           string
+	NodeName        string
+	MarkedForDelete bool
```

Comment:
We can make this a lower case variable and have a function to check IsMarkedForDelete() so that accidently not get set in dp_win.go